cpp/src/table/legacy/table.cpp

```diff
@@ -92,6 +92,16 @@ void table::destroy(void) {
   }
 }
 
+table table::select(std::vector<gdf_size_type> const& column_indices) const {
+    CUDF_EXPECTS(column_indices.size() <= num_columns(), "Requested too many columns.");
+
+    std::vector<gdf_column*> desired_columns;
+    for(auto index : column_indices){
+        desired_columns.push_back(_columns[index]);
```

Comment:
Let's add some bounds checking here. Make sure to update the documentation to indicate that it throws `std::out_of_range` if an index is out of range. 
```suggestion
        desired_columns.push_back(_columns.at(index));
```